compiling error

thanks for all the work done.  The validation is ok But I don't get it to compile. There seems to be an issue with the fan.h or fan.cpp 

HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
Dependency Graph
|-- SPI @ 1.0
|-- Ticker @ 1.0
|-- ESP8266WiFi @ 1.0
|-- ESP8266mDNS @ 1.2
Compiling .pioenvs/fancontrol/src/esphome/components/api/api_connection.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/api/api_frame_helper.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/api/api_frame_helper_plaintext.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/api/api_pb2.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/api/api_pb2_service.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/api/api_server.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/api/list_entities.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/api/proto.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/api/subscribe_state.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/button/button.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/cc1101/CC1101.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/cc1101/IthoCC1101.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/cc1101/fan.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/esp8266/core.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/esp8266/gpio.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/esp8266/helpers.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/esp8266/preferences.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/esphome/ota/ota_esphome.cpp.o
In file included from src/esphome/components/cc1101/fan.cpp:2:
src/esphome/components/cc1101/fan.h:18:36: error: 'set' in namespace 'std' does not name a template type
   18 |   void set_preset_modes(const std::set<std::string> &presets) { this->preset_modes_ = presets; }
      |                                    ^~~
src/esphome/components/cc1101/fan.h:4:1: note: 'std::set' is defined in header '<set>'; did you forget to '#include <set>'?
    3 | #include "esphome/core/hal.h"
  +++ |+#include <set>
    4 | #include "esphome/core/gpio.h"
src/esphome/components/cc1101/fan.h:18:39: error: expected ',' or '...' before '<' token
   18 |   void set_preset_modes(const std::set<std::string> &presets) { this->preset_modes_ = presets; }
      |                                       ^
src/esphome/components/cc1101/fan.h:35:8: error: 'set' in namespace 'std' does not name a template type
   35 |   std::set<std::string> preset_modes_{};
      |        ^~~
src/esphome/components/cc1101/fan.h:35:3: note: 'std::set' is defined in header '<set>'; did you forget to '#include <set>'?
   35 |   std::set<std::string> preset_modes_{};
      |   ^~~
src/esphome/components/cc1101/fan.h: In member function 'void esphome::cc1101fan::CC1101Fan::set_preset_modes(int)':
src/esphome/components/cc1101/fan.h:18:71: error: 'class esphome::cc1101fan::CC1101Fan' has no member named 'preset_modes_'; did you mean 'const char* esphome::fan::Fan::preset_mode_'? (accessible via 'const char* esphome::fan::Fan::get_preset_mode() const')
   18 |   void set_preset_modes(const std::set<std::string> &presets) { this->preset_modes_ = presets; }
      |                                                                       ^~~~~~~~~~~~~
      |                                                                       get_preset_mode()
src/esphome/components/cc1101/fan.h:18:87: error: 'presets' was not declared in this scope
   18 |   void set_preset_modes(const std::set<std::string> &presets) { this->preset_modes_ = presets; }
      |                                                                                       ^~~~~~~
Compiling .pioenvs/fancontrol/src/esphome/components/fan/fan.cpp.o
Compiling .pioenvs/fancontrol/src/esphome/components/logger/logger.cpp.o
*** [.pioenvs/fancontrol/src/esphome/components/cc1101/fan.cpp.o] Error 1
========================= [FAILED] Took 15.40 seconds =========================

--- a/components/cc1101/fan.h
+++ b/components/cc1101/fan.h
@@ -1,3 +1,4 @@
+#include <set>
 #include "esphome/components/fan/fan.h"
 #include "esphome/core/component.h"
 #include "esphome/core/hal.h"

